packages/noya-designsystem/src/components/GridView.tsx

```diff
@@ -64,37 +71,76 @@ const ItemDescription = styled.span(({ theme }) => ({
   textOverflow: 'ellipsis',
 }));
 
+type MenuItemType = 'delete' | 'copy';
+
 interface ItemProps {
-  children?: ReactNode;
-  title: string;
+  title?: string;
   subtitle?: string;
   selected: boolean;
   onClick?: (event: React.MouseEvent) => void;
+  children?: ReactNode;
+  menuItems?: ContextMenu.MenuItem<MenuItemType>[];
+  onSelectMenuItem?: (value: MenuItemType) => void;
+  onContextMenu?: () => void;
 }
 
-function GridItem({ children, title, subtitle, selected, onClick }: ItemProps) {
-  const handleClick = useCallback(
-    (event: React.MouseEvent) => {
-      event.stopPropagation();
-      event.preventDefault();
-
-      onClick?.(event);
-    },
-    [onClick],
-  );
-
-  return (
-    <SwatchContainer>
-      <ItemContainer onClick={handleClick} selected={selected}>
-        {children}
-      </ItemContainer>
-      <Spacer.Vertical size={8} />
-      {/* Use an empty string to ensure element height */}
-      <ItemTitle>{title || ' '}</ItemTitle>
-      <ItemDescription>{subtitle || ' '}</ItemDescription>
-    </SwatchContainer>
-  );
-}
+const GridItem = memo(
+  forwardRef(function GridItem(
+    {
+      title,
+      subtitle,
+      selected,
+      onClick,
+      children,
+      menuItems,
+      onSelectMenuItem,
+      onContextMenu,
+    }: ItemProps,
+    forwardedRef: ForwardedRef<HTMLLIElement>,
+  ) {
+    const handleClick = useCallback(
+      (event: React.MouseEvent) => {
+        event.stopPropagation();
+        event.preventDefault();
+
+        onClick?.(event);
+      },
+      [onClick],
+    );
+
+    const renderContent = (ref: Ref<HTMLLIElement>) => {
```

Comment:
We used the `renderContent` function to help handle sorting in the ListView, but since we don't do sorting here, we don't need to wrap everything in a function

Reply:
Solved it, I had the wrong type for the ref